### DISCUSSION TITLE
Yarpserver results in Access Denied

### DISCUSSION THREAD

**Discussion by dariosortino** [robotology/community#626] [Q&A] [answered]
Hello!
After having enabled the ICUB_HEAD module of the superbuild I encountered a weird behavior when I launch yarpserver (I'm on Windows)

![image](https://github.com/robotology/community/assets/70700623/3144ef63-57f1-43d6-b34a-a6dc1d34339f)

Talking with @traversaro it seems that is an OS-related problem, so he suggested me to trace the events that are happening in the OS. I used the [Process Monitor](https://learn.microsoft.com/en-us/sysinternals/downloads/procmon) to see that in fact the Access Denied is from yarpserver.

![image](https://github.com/robotology/community/assets/70700623/a54769b2-e1d2-4075-a384-6c38807d367d)

![image](https://github.com/robotology/community/assets/70700623/02d2b552-1181-4f5d-af24-0ff3c647075a)

![image](https://github.com/robotology/community/assets/70700623/1cda675b-c1ad-46c5-9119-cf0c882b97b0)

Any hints on what it could be?

**Comment by traversaro** [comment]
It seems that `yarpserver` does not even start, `cmd.exe` returns the error even before starting `yarpserver.exe` . Can you try to delete the file and install again the superbuild (no need to recompiled everything by deleting the build, just reinstalling on the existing build is probably enough).

**Reply by dariosortino** [reply]
Any reference on how to just install?

**Reply by traversaro** [reply]
If you do not remove the build directory, re-running the compile step of the superbuild will just install all the files again (see https://github.com/robotology/robotology-superbuild/blob/master/doc/conda-forge.md#compile-the-robotology-superbuild), so:
~~~
cd robotology-superbuild
cd build
cmake --build . --config Release 
~~~

**Reply by dariosortino** [reply]
Unfortunately, removing yarpserver.exe and installing again didn't solve the problem.

**Reply by traversaro** [reply]
Interesting. What is the output of:
~~~
type C:\Users\dsortino\Desktop\GithubProj\robotology-superbuild\build\install\bin\yarpserver.exe
~~~
? 

Note that if `yarpserver` does not work, you can try to use `yarp server`, with a space between `yarp` and `server`.

**Reply by dariosortino** [reply]
This is the output

![image](https://github.com/robotology/community/assets/70700623/c1444b6c-bf2e-4031-b4d1-1f07c9875ff3)

Surprisingly enough, with `yarp server` everything seems running fine

**Reply by traversaro** [reply, marked as answer]
> This is the output
> 
> ![image](https://user-images.githubusercontent.com/70700623/239897281-c1444b6c-bf2e-4031-b4d1-1f07c9875ff3.png)
> 
> Surprisingly enough, with `yarp server` everything seems running fine

Great, so at least you are able to test what you needed to do by running yarpserver? I have a theory that the problem is related to the length of the path. Can you try to copy `C:\Users\dsortino\Desktop\GithubProj\robotology-superbuild\build\install` to `C:\robsubinstall` and try to use that (i.e. changing the `setup.bat` that you invoke to the one in `C:\robsubinstall\share\robotology-superbuild\setup.bat`)?

**Reply by pattacini** [reply]
Any update about that @dariosortino?

**Reply by dariosortino** [reply]
`yarpserver` started working fine out of nowhere

**Reply by PeterBowman** [reply]
Since the "access denied" event originated in the file system during a "CreateFile" operation (see attached screenshots above), I was wondering whether YARP, due to `--write`, was trying to create a configuration file (yarp.conf) and failed in the process, perhaps due to a lack of permissions in the default location.
